src/module_1/module_1_meteo_api.py

```diff
@@ -1,7 +1,227 @@
+    df_city_climate["city"] = city
+
+    # Save dataframe as csv file
+    data_path = os.getcwd() + "/data"
+    if not os.path.exists(data_path):
+        os.makedirs(data_path)
+
+    save_path = f"{data_path}/{str.lower(city)}_climate.csv"
+    df_city_climate.to_csv(save_path, index=False)
+
+    return df_city_climate
+
+
+def calculate_meteo_mean_max_min_values(df_meteo_0, variables):
+    df_meteo = df_meteo_0.set_index("time")
+
+    for var in variables:
+        columns_var = [col for col in df_meteo.columns if var in col]
+
+        # Get mean, max and mean values of all models available per variable.
+        # Join them together in a single dataframe
+
+        df_join = pd.concat(
+            [
+                pd.DataFrame(
+                    df_meteo[columns_var].mean(axis=1), columns=[var + "_avg"]
+                ),
+                pd.DataFrame(df_meteo[columns_var].max(axis=1), columns=[var + "_max"]),
+                pd.DataFrame(df_meteo[columns_var].min(axis=1), columns=[var + "_min"]),
+            ],
+            axis=1,
+        )
+
+        df_meteo = pd.concat([df_meteo, df_join], axis=1)
+
+    # Select only the columns of interest
+    columns_final = ["time", "city"] + [
+        col
+        for col in df_meteo.columns
+        if col.endswith("_avg") or col.endswith("_max") or col.endswith("_min")
+    ]
+    df_meteo = df_meteo.reset_index()[columns_final]
+
+    return df_meteo
+
+
+def process_data_meteo(cities, variables):
+    # Connect to meteo api or load csv file with data for each city. Concat all info in a single dataframe
+    list_all_cities = []
+
+    for city in cities:
+        data_path = f"{os.getcwd()}/data/{str.lower(city)}_climate.csv"
+
+        # Check if data is available or connect to API to download it
+        if os.path.isfile(data_path):
+            df_city_0 = pd.read_csv(data_path, index_col=False)
+        else:
+            df_city_0 = get_data_meteo_api(city, variables)
+
+        # Call function to calculate mean, max and min values per meteo variable
+        df_city = calculate_meteo_mean_max_min_values(
+            df_meteo_0=df_city_0, variables=variables
+        )
+        columns = df_city.columns
+
+        # Aggregate monthly data
+        df_city["time"] = pd.to_datetime(df_city["time"], format="%d/%m/%Y")
+        df_city["time"] = df_city["time"].dt.to_period("Y")
+        df_city = df_city.groupby(["time", "city"]).agg(["mean"]).reset_index()
+
+        # Reset column names to avoid 'mean' header
+        df_city.columns = columns
+
+        # Append resulting dataframe in list
+        list_all_cities.append(df_city)
+
+    df_all_cities = pd.concat(list_all_cities)
+
+    return df_all_cities
+
+
+def plot_meteo_variables(cities, variables):
+    df_data = process_data_meteo(cities, variables)
+
+    for city in cities:
+        for var in variables:
+            df = df_data[df_data["city"] == city]
+
+            x = df["time"].astype(str).values
```

Comment:
instead of creating new variables, create new columns on the data frame (`df["y_mean"] = df[var + "_avg"]`)